AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/deployments/DeploymentsRest.java

```diff
@@ -116,16 +114,41 @@ public Response getDeployments(@ApiParam("Filters") @QueryParam("filters") Strin
             String detail = "example: [{\"name\":\"Application\",\"comp\":\"eq\",\"val\":\"Latest\"},{\"name\":\"Id\",\"comp\":\"eq\",\"val\":\"25\"}]";
             return Response.status(Status.BAD_REQUEST).entity(new ExceptionDto(msg, detail)).build();
         }
-        CommonFilterService.SortingDirectionType sortingDirectionType = null;
-        if (sortDirection != null) {
-            sortingDirectionType = CommonFilterService.SortingDirectionType.valueOf(sortDirection);
-        }
+        CommonFilterService.SortingDirectionType sortingDirectionType = getSortingDirectionType(sortDirection);
         LinkedList<CustomFilter> filters = createCustomFilters(filterDTOs);
         Tuple<Set<DeploymentEntity>, Integer> filteredDeployments = deploymentBoundary.getFilteredDeployments(offset, maxResults, filters, colToSort, sortingDirectionType, null);
         List<DeploymentDTO> deploymentDTOs = createDeploymentDTOs(filteredDeployments);
         return Response.status(Status.OK).header("X-Total-Count", filteredDeployments.getB()).entity(deploymentDTOs).build();
     }
 
+    @GET
+    @Path("/filterCsvExport")
+    @Produces({"text/comma-separated-values", "text/csv"})
+    @ApiOperation(value = "returns a csv containing all Deployments matching the list of json filters - used by Angular", nickname = "CSV export")
+    public List<DeploymentDTO> getDeploymentsAsCsv(@ApiParam("Filters") @QueryParam("filters") String jsonListOfFilters,
```

Comment:
This should not be needed, the getDeployments endpoint should alread return CSV if the client sets the `Accept` request header to `text/csv`.

Reply:
You are right. It's a remnant from my effort to convince swagger to display both endpoints under the same path.